example/nomad/tempo-distributed/tempo.hcl

```diff
@@ -0,0 +1,546 @@
+        port     = "grpc"
+        type     = "grpc"
+        interval = "20s"
+        timeout  = "1s"
+        grpc_use_tls = false
+        tls_skip_verify = true
+      }
+    }
+
+    task "distributor" {
+      driver       = "docker"
+      user         = "nobody"
+      kill_timeout = "90s"
+
+      config {
+        image = "grafana/tempo:${var.version}"
+        ports = [
+          "http",
+          "grpc",
+          "otpl",
+        ]
+
+        args = [
+          "-target=distributor",
+          "-config.file=/local/config.yml",
+          "-config.expand-env=true",
+        ]
+      }
+
+      template {
+        data        = file("config.yml")
+        destination = "local/config.yml"
+      }
+
+      template {
+        data = <<-EOH
+        S3_ACCESS_KEY_ID=${var.s3_access_key_id}
+        S3_SECRET_ACCESS_KEY=${var.s3_secret_access_key}
+        EOH
+
+        destination = "secrets/s3.env"
+        env         = true
+      }
+
+      resources {
+        cpu        = 200
+        memory     = 128
+        memory_max = 1024
+      }
+    }
+  }
+  group "querier" {
+    count = 1
+
+    network {
+      port "http" {}
+      port "grpc" {}
+    }
+
+    service {
+      name = "tempo-querier"
+      port = "http"
+      tags = [
+        "traefik.enable=true",
+        "traefik.http.routers.tempo-querier.rule=Host(`tempo-querier.service.consul`)"
+      ]
+      check {
+        name     = "Tempo querier"
+        port     = "http"
+        type     = "http"
+        path     = "/ready"
+        interval = "50s"
+        timeout  = "1s"
+      }
+    }
+
+    service {
+      name = "tempo-querier-grpc"
+      port = "grpc"
+      tags = [
+        "grpc",
+        "traefik.enable=true",
+        "traefik.http.routers.tempo-querier-grpc.rule=Host(`tempo-querier-grpc.service.consul`)",
+        "traefik.http.services.tempo-querier-grpc.loadbalancer.server.scheme=h2c"
+      ]
+      check {
+        port     = "grpc"
+        type     = "grpc"
+        interval = "20s"
+        timeout  = "1s"
+        grpc_use_tls = false
+        tls_skip_verify = true
+      }
+    }
+
+    task "querier" {
+      driver       = "docker"
+      user         = "nobody"
+      kill_timeout = "90s"
+
+      config {
+        image = "grafana/tempo:${var.version}"
+        ports = [
+          "http",
+          "grpc",
+        ]
+
+        args = [
+          "-target=querier",
+          "-config.file=/local/config.yml",
+          "-config.expand-env=true",
+        ]
+      }
+
+      template {
+        data        = file("config.yml")
+        destination = "local/config.yml"
+      }
+
+      template {
+        data = <<-EOH
+        S3_ACCESS_KEY_ID=${var.s3_access_key_id}
+        S3_SECRET_ACCESS_KEY=${var.s3_secret_access_key}
+        EOH
+
+        destination = "secrets/s3.env"
+        env         = true
+      }
+
+      resources {
+        cpu        = 200
+        memory     = 128
+        memory_max = 2048

```

Comment:
do we want to specify a max? I guess as an example that's fine, but resource usage of Tempo can cover a huge range depending on usage patterns.

Reply:
`memory` value is how much RAM will nomad reserve for the task, `memory_max` value is how much task can expand if it goes above reserved memory (memory oversubscription). `memory_max` values can be removed, in that case value of `memory` will be max value. I would leave it as is, since this is an example, pepole can decide what works best for them. I will update README.md to address this.